### PR TITLE
Clean up build to prep for 2.12

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- Adds `codecov.io` badges and support to the build
- Adds a Gitter storehaus channel webhook for build status updates -> Gitter
- enables caching in the travis build
- adds build, scaladex, gitter and code coverage badges to the README
- upgrades sbt to 0.13.13 and fixes deprecation warnings
- upgrades plugins like mima, jmh to latest versions
- kills old, deprecated doc tasks (I'll add a microsite soon)

This should get us ready for 2.12.0 once the relevant deps are in place.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,15 +2,15 @@ resolvers ++= Seq(
...
-addSbtPlugin("org.scoverage" % "sbt-coveralls" % "1.0.0.BETA1")
-addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.0.4")
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "1.0")
-addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.2.2")
+addSbtPlugin("com.eed3si9n"       % "sbt-assembly"    % "0.11.2")
+addSbtPlugin("com.eed3si9n"       % "sbt-unidoc"      % "0.3.3")
+addSbtPlugin("com.fortysevendeg"  % "sbt-microsites"  % "0.3.3")
+addSbtPlugin("com.github.gseitz"  % "sbt-release"     % "1.0.0")
+addSbtPlugin("com.jsuereth"       % "sbt-pgp"         % "1.0.0")
+addSbtPlugin("com.typesafe"       % "sbt-mima-plugin" % "0.1.11")
```
Comment (johnynek) [inline project/plugins.sbt]: can we bump this to `0.1.12`? I think that's the latest with a bugfix we found.

**Review by johnynek — CHANGES_REQUESTED** [review]
looks good other than the minor change on mima.

**Comment by sritchie** [comment]
@johnynek done!

**Comment by johnynek** [comment]
looks like it does not build.

**Comment by sritchie** [comment]
Will fix when I get on a real computer

**Comment by sritchie** [comment]
Okay @johnynek, NOW we should be good. That version bump killed some keys that we were importing (but not using). All tidied up now.

**Comment by sritchie** [comment]
Build just needed a kick, @johnynek. Good to go now when tests pass.

**Comment by sritchie** [comment]
Hmm, seems to be getting stuck after the kafka tests.

**Comment by sritchie** [comment]
Nice @johnynek... looks like that upgrade DID catch two more binary incompatibilities. We have a TON of those now, btw. Definite version bump required.

File: build.sbt [hunk 2]
```diff
@@ -59,7 +55,9 @@ val ignoredABIProblems = {
     exclude[MissingMethodProblem]("com.twitter.storehaus.asynchbase.AsyncHBaseStringStore" +
       ".futurePool"),
     exclude[MissingMethodProblem]("com.twitter.storehaus.asynchbase.AsyncHBaseStore.futurePool"),
-    exclude[MissingMethodProblem]("com.twitter.storehaus.ReadThroughStore.mutex")
+    exclude[MissingMethodProblem]("com.twitter.storehaus.ReadThroughStore.mutex"),
+    exclude[MissingClassProblem]("com.twitter.storehaus.kafka.JavaFutureToTwitterFutureConverter$Closed$"),
```
Comment (johnynek) [inline build.sbt:59]: I think this code got migrated to bijection anyway, so we can use that on the next bump.

**Review by johnynek — APPROVED** [review]
:+1:

File: build.sbt (hunk 2 above)
Comment (johnynek) [reply build.sbt:59]: Added #337

**Pull request merged**